### ISSUE TITLE
How much memory for training?

### ISSUE THREAD

**Issue by duynn912** [ucbdrive/few-shot-object-detection#7]
Dear authors,

Thank you for your shared source of code!

I have a problem with memory. I was wondering if you could please tell me how much memory of GPU for training. Btw, I only have 1 GPU 12Gb.

Best regards.

**Comment by thomasehuang** [comment]
Thank you for your interest.

We train all our models on 8 GPUs with a batch size of 16 (2 images on each GPU). Since you only have one GPU, the current setup will try to put 16 images on one single GPU, which would be too much memory. You should modify the config files to use a batch size of 2, while reducing the learning rate by 8 times and increasing the total number of training iterations by 8 times. 12Gb of memory should be more than enough.

**Comment by xinw1012** [comment]
Thanks for your interest in our work. 

The training schedules in the provided configurations are based on 8 GPUs. We used 8  GeForce RTX 2080 Ti in our experiments which has 11Gb memory for each GPU. 

To run on a single GPU, you can adjust the batch size and the learning rate according to the linear scaling rule described here https://arxiv.org/abs/1706.02677. That is, `when the minibatch size is multiplied by k, multiply the learning rate by k.`  The mask-rcnn benchmark codebase has an example configuration. You can also take a look at here . https://github.com/facebookresearch/Detectron/blob/master/configs/getting_started/tutorial_1gpu_e2e_faster_rcnn_R-50-FPN.yaml#L14-L30

Hope this helps!

**Comment by xinw1012** [comment]
haha both of us replied at the same time. lol

**Comment by duynn912** [comment]
Dear @xinw1012 and @thomasehuang,

Thank you very much for your quick response!
It is really coincidental that both of you respond at the same time. This is the first time, I meet this event. I hope your work will be accepted in a top conference in case of it has been submitted or in the future!

**Comment by thomasehuang** [comment]
Thank you very much!

**Comment by duynn912** [comment]
> Sorry to bother you, but have you reproduced the results in the paper with a GPU

Yup, I have reproduce with 1 GPU by following the author's guidelines on above comments. Hope it good for you!

**Comment by xiaoliqiangxxqzz** [comment]
> > Sorry to bother you, but have you reproduced the results in the paper with a GPU
> 
> Yup, I have reproduce with 1 GPU by following the author's guidelines on above comments. Hope it good for you!

Can you share me your parameters (such as batchsize, lr, etc.), I have done a lot of attempts on voc spilt1, the first stage uses the author's trained model:
1. Use a GPU to train the second stage on the default configuration;
2.gpu                8 -> 1
    batchsize      16->2
    lr                  0.001 -> 0.000125
    step              3000 -> 24000
    max_iter       4000 ->32000

**Comment by hero-y** [comment]
Hello, I trained the same config file twice, but got two very different results. For example: in VOC Split1 1shot, they are 41.2 nAP50 and 38.7 nAP50. In VOC Split1 3shot, they are 48.4 nAP50 and 45.6 nAP50. The results look very unstable. Have you tried to run the same config file multiple times? Are the results stable?

**Comment by salehnia** [comment]
Hello, Can you share me your parameters for 1 GPU in training coco dataset?
